File: app/src/main/java/br/com/nicetuber/feature/home/HomeActivity.java
package br.com.nicetuber.feature.home;

import android.support.design.widget.Snackbar;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.SearchView;
import android.view.Menu;
import android.view.MenuItem;

import java.util.List;

import br.com.nicetuber.R;
import br.com.nicetuber.base.BaseActivity;
import br.com.nicetuber.databinding.ActivityHomeBinding;
import br.com.nicetuber.model.ChannelSearched;
import br.com.nicetuber.util.UIListeners;

public class HomeActivity extends BaseActivity<ActivityHomeBinding, HomePresenter>
        implements IHome.View, UIListeners.OnClickListener, SearchView.OnQueryTextListener {

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.options_menu, menu);

        MenuItem searchItem = menu.findItem(R.id.action_search);
        SearchView searchView = (SearchView) searchItem.getActionView();
        searchView.setOnQueryTextListener(this);
        searchItem.setOnActionExpandListener(new MenuItem.OnActionExpandListener() {
            @Override
            public boolean onMenuItemActionExpand(MenuItem menuItem) {
                return true;
            }

            @Override
            public boolean onMenuItemActionCollapse(MenuItem menuItem) {
                HomeActivity.super.closeKeyboard();
                return true;
            }
        });

        return true;
    }

    @Override
    protected void injectDagger() {
        getAppComponent().inject(this);
    }

    @Override
    protected int getXmlLayout() {
        return R.layout.activity_home;
    }

    @Override
    protected void attachView() {
        presenter.attachView(this);
    }

    @Override
    protected void detachView() {
        presenter.detachView();
    }

    @Override
    public void showList(List<ChannelSearched> response) {
        super.closeKeyboard();

        RecyclerView recyclerView = binding.rvChannelList;
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        recyclerView.setHasFixedSize(true);

        HomeChannelAdapter adapter = new HomeChannelAdapter(response, this);
        recyclerView.setAdapter(adapter);
    }

    @Override
    public void showErrorMessage(String message) {
        Snackbar.make(binding.clActivityMain, message, Snackbar.LENGTH_LONG).show();
    }

    @Override
    public boolean onQueryTextSubmit(String query) {
        presenter.searchChannel(query);
        return true;
    }

    @Override
    public boolean onQueryTextChange(String s) {
        return false;
    }

    @Override
    public void onClick(ChannelSearched channelSearched) {
        HomeRouter.goToChannelDetailsView(this, channelSearched);
    }
}

File: app/src/main/java/br/com/nicetuber/feature/home/HomeActivity.kt
package br.com.nicetuber.feature.home

import android.support.design.widget.Snackbar
import android.support.v7.widget.LinearLayoutManager
import android.support.v7.widget.SearchView
import android.view.Menu
import android.view.MenuItem
import android.view.View
import br.com.nicetuber.R
import br.com.nicetuber.base.BaseActivity
import br.com.nicetuber.model.ChannelSearched
import br.com.nicetuber.util.UIListeners
import kotlinx.android.synthetic.main.activity_home.*

class HomeActivity : BaseActivity<HomePresenter>(),
        IHome.View, UIListeners.OnClickListener, SearchView.OnQueryTextListener {

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.options_menu, menu)

        val searchItem = menu.findItem(R.id.action_search)
        val searchView = searchItem.actionView as SearchView
        searchView.setOnQueryTextListener(this)
        searchItem.setOnActionExpandListener(object : MenuItem.OnActionExpandListener {
            override fun onMenuItemActionExpand(menuItem: MenuItem): Boolean {
                return true
            }

            override fun onMenuItemActionCollapse(menuItem: MenuItem): Boolean {
                super@HomeActivity.closeKeyboard()
                return true
            }
        })

        return true
    }

    override fun injectDagger() {
        appComponent?.inject(this)
    }

    override fun getXmlLayout(): Int {
        return R.layout.activity_home
    }

    override fun attachView() {
        presenter.attachView(this)
    }

    override fun detachView() {
        presenter.detachView()
    }

    override fun showList(response: List<ChannelSearched>) {
        super.closeKeyboard()

        rv_channel_list.layoutManager = LinearLayoutManager(this)
        rv_channel_list.setHasFixedSize(true)

        val adapter = HomeChannelAdapter(response, this)
        rv_channel_list.adapter = adapter
    }

    override fun showErrorMessage(message: String) {
        Snackbar.make(cl_activity_main as View, message, Snackbar.LENGTH_LONG).show()
    }

    override fun onQueryTextSubmit(query: String): Boolean {
        presenter.searchChannel(query)
        return true
    }

    override fun onQueryTextChange(s: String): Boolean {
        return false
    }

    override fun onClick(channelSearched: ChannelSearched) {
        HomeRouter.goToChannelDetailsView(this, channelSearched)
    }
}
